sktime/utils/validation/forecasting.py

```diff
@@ -455,3 +455,35 @@ def check_regressor(regressor=None, random_state=None):
             )
         regressor = clone(regressor)
     return regressor
+
+
+def check_interval_df(interval_df, index_to_match):
```

Comment:
some of this is redundant with

```python
from sktime.datatypes import check_is_mtype

check_is_mtype(interval_df, "pred_interval")
```

(I think all except the "index to match" check)

Reply:
I will check that out and replace these lines, good tip

Reply:
After looking into it, I am leaning towards leaving the checks as is. I would typically want to reuse `check` style code but the existing code raises very informative errors and the function needs to exist regardless, at least to check the column name (used in the plotting function) and the matching index as you mentioned. Is that acceptable?

Reply:
sure, but that would mean you don´t find `check_is_mtype` informative of the same degree, which should be improved then, separately.

Reply:
I was not fully aware of the usefulness of `return_metadata=True`, now that I see it I've added it in. I kept checks for index matching, exact column name as used in the plot function ("Coverage"), and that there is only one pair of interval series. Should be good to go if this passes tests.

FYI you mentioned a "non blocking" comment but it appears this PR can't be completed until all comments are resolved. It's very possible I misunderstood what you meant though.

Reply:
in the end, a core developer needs to merge it either way